### PR TITLE
Portability fixes for SchemaBuilder.

### PR DESCRIPTION
**Pull request by nkovacs**
Removed length from integer and boolean types. This is not supported on PostgreSQL,
and is only used for display in MySQL, not input validation.

Removed length from text and binary type. This is not supported on most RDBMS's and ignored by others.

Changed binary mapping to `varbinary(max)` on MSSQL instead of `binary(1)` (which requires a length and is fixed length) to better match others.

Changed timestamp mapping to datetime on MSSQL. Timestamp is a unique number on MSSQL, not a timestamp.

Removed precision and scale parameters from floating point types. This is not supported by PostgreSQL, among others.

Renamed timestamp length parameter to precision. The precision parameter is not supported on all RDBMS's, and may throw an error. This can be fixed either by removing the precision parameter, or overriding the SchemaBuilder of those RDBMS's to ignore the precision parameter.
**This still needs to be fixed**, though  it's not a problem on MySQL and PostgresQL.

Since the release date of 2.0.6 is close, I've opted to remove these parameters. Removing them later would require a BC break. Adding them in later (though I don't think that's a good idea) would not.
I think it would be a better idea to add a `length`, `precision` etc. function, so that if you know you're using a DB that supports these, you can do:

```
    'foo' => Schema::integer()->length(10)
    'bar' => Schema::float()->precision(4)->scale(4)
```

See #9102

### CODE REVIEW

File: framework/db/mssql/QueryBuilder.php [hunk 1]
```diff
@@ -34,10 +34,10 @@ class QueryBuilder extends \yii\db\QueryBuilder
         Schema::TYPE_DOUBLE => 'float',
         Schema::TYPE_DECIMAL => 'decimal',
         Schema::TYPE_DATETIME => 'datetime',
-        Schema::TYPE_TIMESTAMP => 'timestamp',
+        Schema::TYPE_TIMESTAMP => 'datetime',
         Schema::TYPE_TIME => 'time',
         Schema::TYPE_DATE => 'date',
-        Schema::TYPE_BINARY => 'binary(1)',
```
Comment (samdark) [inline framework/db/mssql/QueryBuilder.php]: https://msdn.microsoft.com/en-us/library/ms188362.aspx

Fixed length makes more sense, doesn't it?

Comment (nkovacs) [inline framework/db/mssql/QueryBuilder.php]: TYPE_BINARY is blob elsewhere, which is variable length.

**Comment by nkovacs** [comment]
Ok, so QueryBuilder::getColumnType will remove the lengths if the value in typemap does not have a length, so `text(255) NOT NULL` is converted to `text NOT NULL`. This means the DB won't throw errors,  so there's no need to remove the parameter where some RDBMS's support it.

**Comment by samdark** [comment]
@nkovacs yep. Can you adjust pull request accordingly?

**Comment by nkovacs** [comment]
Ok, I'll put the length back where it makes sense.

**Comment by samdark** [comment]
btw., changing public or protected method signature is always a backwards compatibility break. It doesn't matter if it's removing parameters or adding parameters.

**Comment by nkovacs** [comment]
I still have to check how float works, since the typemap doesn't have a length.

> btw., changing public or protected method signature is always a backwards compatibility break. It doesn't matter if it's removing parameters or adding parameters.

Yeah, but if it has a default value, only subclasses are affected, and you've done that before in point releases. It's still a smaller break.

**Comment by nkovacs** [comment]
Ok, so the problem with float and double is that the only typemap that includes a length is cubrid's, but cubrid only accepts a single parameter, not two. So there's no point in having precision and scale for those two.

**Comment by nkovacs** [comment]
I had to comment out two tests on Cubrid, because they won't work without the float parameter.

File: framework/db/mssql/QueryBuilder.php (hunk 1 above)
Comment (samdark) [inline framework/db/mssql/QueryBuilder.php]: OK.

**Comment by samdark** [comment]
@nkovacs you mean Cubrid doesn't work w/o float parameter?

**Comment by nkovacs** [comment]
I removed the precision and scale parameters from the float function, because only cubrid used it, but cubrid only accepts one parameter, and it might fail if you specify both (e.g. `float(4,4)`), and I think getColumnType will allow that.
Two of the tests tested that the float parameter works, i.e. that `Schema::float(16)` creates a column that's `float(16)`, not the default in typemap (`float(7)`), so those tests cannot pass now. For the other RDBMS's, these tests pass, because the parameters to `Schema::float` are ignored.

**Comment by samdark** [comment]
Then it makes sense to remove these tests instead of commenting them out.

**Comment by nkovacs** [comment]
Those tests also verify that `Schema::TYPE_FLOAT . '(16)'` is converted to `float(16)`, which works on cubrid because the typemap allows it.

**Comment by cebe** [comment]
why did you remove the precision from float and double then?

**Comment by nkovacs** [comment]
Another possibility is to keep the precision and scale parameters, and extend CubridSchemaBuilder so that it ignores scale. Then you can add support for the parameters to other db's later.

File: framework/db/mssql/QueryBuilder.php (hunk 1 above)
Comment (cebe) [inline framework/db/mssql/QueryBuilder.php]: these changes are BC breaking, separated them into another PR and tagged it for 2.1: #9305

Comment (nkovacs) [inline framework/db/mssql/QueryBuilder.php]: Yeah, but this is totally broken.

Comment (cebe) [inline framework/db/mssql/QueryBuilder.php]: sure but if someone was using these in MSSQL we can not just break their code.

Comment (nkovacs) [inline framework/db/mssql/QueryBuilder.php]: I know. I guess MSSQL is not used that often, so it's not that critical.

**Pull request closed without merging**